utils/project/sync.go

```diff
@@ -129,14 +147,14 @@ func syncFiles(projectPath string, projectId string, synctime int64) ([]string,
 	return fileList, modifiedList
 }
 
-func completeUpload(projectId string, files []string, modfiles []string, timestamp int64) {
-	uploadEndUrl := config.PFEApiRoute() + "projects/" + projectId + "/upload/end"
+func completeUpload(projectID string, files []string, modfiles []string, depURL string, timestamp int64) {
+	uploadEndURL := depURL + "projects/" + projectID + "/upload/end"
 
 	payload := &CompleteRequest{FileList: files, ModifiedList: modfiles, TimeStamp: timestamp}
 	jsonPayload, _ := json.Marshal(payload)
 
 	// Make the request to end the sync process.
-	resp, err := http.Post(uploadEndUrl, "application/json", bytes.NewBuffer(jsonPayload))
+	resp, err := http.Post(uploadEndURL, "application/json", bytes.NewBuffer(jsonPayload))
 	fmt.Println("Upload end status:" + resp.Status)
 	if err != nil {
 		panic(err)
```

Comment:
Not strictly part of this PR and I do see the TODO so until we figure out a way to undo/retry this seem ok. Can the IDE handle the panic message ?